src/rgw/rgw_rados.cc

```diff
@@ -1926,17 +1926,17 @@ int RGWRados::Bucket::List::list_objects_ordered(int64_t max_p,
     }
 
     ldout(cct, 20) << "RGWRados::Bucket::List::" << __func__ <<
-      " INFO end of outer loop, trucated=" << truncated <<
+      " INFO end of outer loop, truncated=" << truncated <<
       ", count=" << count << ", attempt=" << attempt << dendl;
 
-    // if we finished listing, or if we're returning at least half the
-    // requested entries, that's enough; S3 and swift protocols allow
-    // returning fewer than max entries
     if (!truncated || count >= (max + 1) / 2) {
+      // if we finished listing, or if we're returning at least half the
+      // requested entries, that's enough; S3 and swift protocols allow
+      // returning fewer than max entries
       break;
-    } else if (attempt > 4 and count >= (max + 7) / 8) {
-      break;
-    } else if (attempt > 8 and count >= 1) {
+    } else if (attempt > 8 && count >= 1) {
```

Comment:
ok

Reply:
Yeah, initially I typed "and" rather than "&&" for some reason, and to be honest, I didn't even know that was legal C++. But it's now in the canonical form.